Toshi/Source/Toshi/File/TTRB.cpp

```diff
@@ -6,28 +6,28 @@ bool Toshi::TTRB::LoadTrb(const char* fn)
 	//TFile* f = TFile::Create(fn, 1);
 
 	TCString str = TCString(fn);
-	Toshi::TNativeFileSystem * fs = Toshi::tnew<Toshi::TNativeFileSystem>(str);
-	Toshi::TNativeFile* f = (Toshi::TNativeFile*)fs->CreateFile(str, 1);
-	FUN_00686920(f);
+	auto fs = Toshi::TFileSystem::CreateNative(str);
+	Toshi::TFile* file = fs->CreateFile(str, 1);
+	FUN_00686920(file);
 	return false;
 }
 
-void Toshi::TTRB::FUN_00686920(TNativeFile* a_pFile)
+void Toshi::TTRB::FUN_00686920(TFile* a_pFile)
 {
+	TASSERT(a_pFile != TNULL, "a_pFile is TNULL")
 	TSF tsf;
-	a_pFile->Read(&tsf, 8);
+	a_pFile->Read(&tsf, sizeof(TSF));
 
-	if (tsf.m_magic == 0x4C465354)
+	if (tsf.m_magic == TMAKEFOUR("TSFL"))
```

Comment:
TMAKEFOUR is a constexpr function so the value will be calculated at compile-time and won't take any time at run-time

Reply:
Thank you that's what i searched for the whole time. Couldn't find it on the internet and i'm too unexperienced in c++ rn.